crates/uv/src/commands/tool/run.rs

```diff
@@ -107,51 +109,92 @@ pub(crate) async fn run(
         return Ok(ExitStatus::Error);
     };
 
-    let (target, args) = command.split();
-    let Some(target) = target else {
-        return Err(anyhow::anyhow!("No tool command provided"));
-    };
+    let (target, args) = {
+        let (target, args) = command.split();
+        let Some(target) = target else {
+            return Err(anyhow::anyhow!("No tool command provided"));
+        };
 
-    let Some(target) = target.to_str() else {
-        return Err(anyhow::anyhow!("Tool command could not be parsed as UTF-8 string. Use `--from` to specify the package name"));
-    };
+        let Some(target) = target.to_str() else {
+            return Err(anyhow::anyhow!("Tool command could not be parsed as UTF-8 string. Use `--from` to specify the package name"));
+        };
 
-    if let Some(ref from) = from {
-        if has_python_script_ext(Path::new(from)) {
-            let package_name = PackageName::from_str(from)?;
-            return Err(anyhow::anyhow!(
-                "It looks you provided a Python script to `--from`, which is not supported\n\n{}{} If you meant to run a command from the `{}` package, use the normalized package name instead to disambiguate, e.g., `{}`",
-                "hint".bold().cyan(),
-                ":".bold(),
-                package_name.cyan(),
-                format!("{} --from {} {}", invocation_source, package_name.cyan(), target),
-            ));
-        }
-    } else {
-        let target_path = Path::new(target);
-        if has_python_script_ext(target_path) {
-            return if target_path.try_exists()? {
-                Err(anyhow::anyhow!(
-                    "It looks you tried to run a Python script at `{}`, which is not supported by `{}`\n\n{}{} Use `{}` instead",
-                    target_path.user_display(),
-                    invocation_source,
-                    "hint".bold().cyan(),
-                    ":".bold(),
-                    format!("uv run {}", target_path.user_display().cyan()),
-                ))
-            } else {
-                let package_name = PackageName::from_str(target)?;
-                Err(anyhow::anyhow!(
-                    "It looks you provided a Python script to run, which is not supported supported by `{}`\n\n{}{} We did not find a script at the requested path. If you meant to run a command from the `{}` package, pass the normalized package name to `--from` to disambiguate, e.g., `{}`",
-                    invocation_source,
+        let (mut target, mut args) = (target, args);
```

Comment:
Kind of gross but: I'm indenting the whole thing, making these `mut` within this closure, and then adding new logic below to override them on confirmation.